default.nix

```diff
@@ -28,62 +28,70 @@ let
   application = import applicationPath;
 
   pkgs = import ./pkgs (with application; {
-    inherit version updateUrl kioskUrl;
     applicationOverlays = application.overlays;
   });
 
   # lib.makeScope returns consistent set of packages that depend on each other (and is my new favorite nixpkgs trick)
   components = with pkgs; lib.makeScope newScope (self: with self; {
 
-    version = application.version;
     inherit updateUrl deployUrl kioskUrl;
+    inherit (application) version safeProductName fullProductName;
 
     greeting = lib.attrsets.attrByPath [ "greeting" ] (label: label) application;
 
+    # Controller
+    playos-controller = import ./controller {
```

Comment:
That name standing alone makes me think of [Brave New World](https://www.cliffsnotes.com/literature/b/brave-new-world/character-analysis/mustapha-mond). :smile: 

It could also be `system-controller`, `os-controller`, `central-controller`, ...